### PR TITLE
fix: encoder with attention

### PR DESCRIPTION
**Pull request by jorisSchaller**
Fix the attention issue where in case of batching, some padding is added and those tokens are not ignored in the attention.

### CODE REVIEW

File: src/mblm/model/mblm.py [hunk 1]
```diff
@@ -643,48 +649,570 @@ def generate(
...
+                - Single tensor: Will be reused across all stages (bypasses embedding lookups).
+                    The tensor must be pre-organized into the hierarchical structure:
+                    (B, P1', P2, ..., Pn, Dn) for multi-stage or (B, L, D) for single-stage.
+                    NOTE: This will produce different results than input_ids because each stage
+                    has its own embedding table.
+                - Sequence of tensors: One tensor per stage (in reverse order: local to global).
+                  This allows exact replication of the input_ids path.
+                  Expected shapes: [local_embs (B, P1', P2, ..., Pn, Dn), ..., global_embs (B, P1', D1)]
+            return_type: What to return - the loss, the logits, the the Hidden_state or the loss and the logits
+            mask: The
```
Comment (jannisborn) [inline src/mblm/model/mblm.py]: doc for the attention mask is missing. Could make sense to rename `mask` to `attention_mask` as there is also `loss_mask`

**Review by jannisborn — CHANGES_REQUESTED** [review]
Great job overall! The forward looks good, but we are not using the mask in the `inputs_embeds` track. Why was this done?

File: src/mblm/model/mblm.py [hunk 2]
```diff
@@ -568,12 +569,12 @@ def forward(
         loss_tensor = loss_tensor[:, :flat_seq_len]
 
         if loss_mask is not None:
-            # potentially apply the loss mask. this does not involve
-            # broadcasting as after slicing above, the mask and the loss tensor
+            # potentially apply the loss attention_mask. this does not involve
```
Comment (jannisborn) [inline src/mblm/model/mblm.py]: revert attention_mask back to mask

File: src/mblm/model/mblm.py [hunk 3]
```diff
@@ -568,12 +569,12 @@ def forward(
...
+            # broadcasting as after slicing above, the attention_mask and the loss tensor
             # have the exact same shape again
             loss_tensor *= loss_mask
 
-        # after applying the mask, some elements might be 0 - they should not be
+        # after applying the attention_mask, some elements might be 0 - they should not be
```
Comment (jannisborn) [inline src/mblm/model/mblm.py]: here we mean the loss mask
